src/app/pages/RewardPage/components/ClaimForms/FeesEarnedClaimRow/index.tsx

```diff
@@ -53,44 +61,115 @@ export const FeesEarnedClaimRow: React.FC<IFeesEarnedClaimRowProps> = ({
 
+  const { value: userCheckpoint } = useCacheCallWithValue<IUserCheckpoint>(
+    'feeSharingProxy',
+    'getNextPositiveUserCheckpoint',
+    {},
+    address,
+    contractAddress,
+    0,
+    MAX_NEXT_POSITIVE_CHECKPOINT,
+  );
+
+  const { checkpointNum, hasFees, hasSkippedCheckpoints } = useMemo(() => {
+    return {
+      checkpointNum: userCheckpoint.checkpointNum,
+      hasFees: userCheckpoint.hasFees,
+      hasSkippedCheckpoints: userCheckpoint.hasSkippedCheckpoints,
+    };
+  }, [userCheckpoint]);
+
   const { value: maxCheckpoints } = useCacheCallWithValue(
     'feeSharingProxy',
     'numTokenCheckpoints',
     -1,
     contractAddress,
   );
-  const { send, ...tx } = useSendContractTx('feeSharingProxy', 'withdraw');
+  const { send: withdraw, ...tx } = useSendContractTx(
+    'feeSharingProxy',
+    'withdraw',
+  );
   const { send: withdrawRBTC, ...tx2 } = useSendContractTx(
     'feeSharingProxy',
     'withdrawRBTC',
   );
+  const { send: withdrawStartingFromCheckpoint, ...tx3 } = useSendContractTx(
+    'feeSharingProxy',
+    'withdrawStartingFromCheckpoint',
+  );
+  const {
+    send: withdrawRBTCStartingFromCheckpoint,
+    ...tx4
+  } = useSendContractTx(
+    'feeSharingProxy',
+    'withdrawRBTCStartingFromCheckpoint',
+  );
 
   const [isSimulating, setIsSimulating] = useState(false);
 
   const isClaimDisabled = useMemo(
     () =>
+      !hasFees ||
       isSimulating ||
       claimFeesEarnedLocked ||
       !bignumber(amountToClaim).greaterThan(0) ||
       assetClaimLocked,
-    [isSimulating, claimFeesEarnedLocked, amountToClaim, assetClaimLocked],
+    [
+      hasFees,
+      isSimulating,
+      claimFeesEarnedLocked,
+      amountToClaim,
+      assetClaimLocked,
+    ],
   );
+  console.log({
+    maxCheckpoints,
+    asset,
+    checkpointNum,
+  });
 
   const onSubmit = useCallback(() => {
-    if (asset === Asset.RBTC) {
-      withdrawRBTC(
-        [0, address],
-        { from: address, gas: gasLimit[TxType.STAKING_REWARDS_CLAIM_RBTC] },
-        { type: TxType.STAKING_REWARDS_CLAIM_RBTC },
-      );
+    if (hasSkippedCheckpoints) {
+      if (asset === Asset.RBTC) {
+        withdrawRBTCStartingFromCheckpoint(
+          [checkpointNum, 0, address],
+          { from: address, gas: gasLimit[TxType.STAKING_REWARDS_CLAIM_RBTC] },
+          { type: TxType.STAKING_REWARDS_CLAIM_RBTC },
+        );
+      } else {
+        withdrawStartingFromCheckpoint(
+          [contractAddress, checkpointNum, maxCheckpoints, address],
+          { from: address, gas: gasLimit[TxType.STAKING_REWARDS_CLAIM] },
+          { type: TxType.STAKING_REWARDS_CLAIM },
+        );
+      }
     } else {
-      send(
-        [contractAddress, maxCheckpoints, address],
-        { from: address, gas: gasLimit[TxType.STAKING_REWARDS_CLAIM] },
-        { type: TxType.STAKING_REWARDS_CLAIM },
-      );
+      if (asset === Asset.RBTC) {
+        withdrawRBTC(
+          [0, address],
```

Comment:
```suggestion
          [maxCheckpoints, address],
```
Should this be `maxCheckpoints`?
<img width="343" alt="image" src="https://github.com/DistributedCollective/Sovryn-frontend/assets/14878713/fac63013-fe38-4a11-8ae9-5dcffd6fc263">